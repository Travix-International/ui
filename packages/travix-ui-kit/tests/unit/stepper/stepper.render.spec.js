import { shallow } from 'enzyme';
import React from 'react';
import Stepper from '../../../components/stepper/stepper';

describe('Stepper', () => {
  describe('#render()', () => {
    it('should render', () => {
      const wrapper = shallow(
        <Stepper
          initValue={2}
          minValue={1}
        />
      );

      expect(wrapper).toMatchSnapshot();
    });

    it('should render with disabled decrement button', () => {
      const wrapper = shallow(
        <Stepper
          initValue={1}
          minValue={1}
        />
      );

      expect(wrapper).toMatchSnapshot();
    });

    it('should render with disabled increment button', () => {
      const wrapper = shallow(
        <Stepper
          initValue={3}
          maxValue={3}
        />
      );

      expect(wrapper).toMatchSnapshot();
    });
  });
});
