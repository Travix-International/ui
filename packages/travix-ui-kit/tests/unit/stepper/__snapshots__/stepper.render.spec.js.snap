// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stepper #render() should render 1`] = `
<div
  className="ui-stepper"
>
  <input
    id="input"
    name="input"
    type="hidden"
    value={2}
  />
  <div
    className="ui-stepper-content"
  >
    <button
      className="icon icon-100 icon-Buttons_arrowMinus ui-stepper-button ui-stepper-button_decrease"
      onClick={[Function]}
      type="button"
    />
    <div
      className="ui-stepper-value"
    >
      2
    </div>
    <div
      className="ui-stepper-shadow"
    />
    <button
      className="icon icon-100 icon-Buttons_arrowPlus ui-stepper-button ui-stepper-button_increase"
      onClick={[Function]}
      type="button"
    />
  </div>
</div>
`;

exports[`Stepper #render() should render with disabled decrement button 1`] = `
<div
  className="ui-stepper"
>
  <input
    id="input"
    name="input"
    type="hidden"
    value={1}
  />
  <div
    className="ui-stepper-content"
  >
    <button
      className="icon icon-100 icon-Buttons_arrowMinus ui-stepper-button ui-stepper-button_decrease disabled"
      onClick={[Function]}
      type="button"
    />
    <div
      className="ui-stepper-value"
    >
      1
    </div>
    <div
      className="ui-stepper-shadow"
    />
    <button
      className="icon icon-100 icon-Buttons_arrowPlus ui-stepper-button ui-stepper-button_increase"
      onClick={[Function]}
      type="button"
    />
  </div>
</div>
`;

exports[`Stepper #render() should render with disabled increment button 1`] = `
<div
  className="ui-stepper"
>
  <input
    id="input"
    name="input"
    type="hidden"
    value={3}
  />
  <div
    className="ui-stepper-content"
  >
    <button
      className="icon icon-100 icon-Buttons_arrowMinus ui-stepper-button ui-stepper-button_decrease"
      onClick={[Function]}
      type="button"
    />
    <div
      className="ui-stepper-value"
    >
      3
    </div>
    <div
      className="ui-stepper-shadow"
    />
    <button
      className="icon icon-100 icon-Buttons_arrowPlus ui-stepper-button ui-stepper-button_increase disabled"
      onClick={[Function]}
      type="button"
    />
  </div>
</div>
`;
